reference-implementation/lib/writable-stream.js

```diff
@@ -533,6 +541,24 @@ function IsWritableStreamDefaultWriter(x) {
   return true;
 }
 
+function WritableStreamDefaultWriterEnsureReadyPromiseRejectedWith(stream, error, isPending) {
```

Comment:
It seems nicer to have this recompute WritableStreamIsReadyForWrites each time. Is passing it in just an optimization, or is it important for correctness (e.g. computing it early in the function)?